app/en/home/build-tools/secure-your-mcp-server/page.mdx

```diff
@@ -0,0 +1,480 @@
+
+<GuideOverview>
+<GuideOverview.Outcomes>
+
+Add [MCP compliant OAuth 2.1 front-door authentication](https://modelcontextprotocol.io/specification/2025-06-18/basic/authorization) to your HTTP MCP server to enable tool-level authorization and secrets.
+
+</GuideOverview.Outcomes>
+
+<GuideOverview.Prerequisites>
+
+- An existing MCP server created with `arcade new` (see [Create an MCP Server](/home/build-tools/create-a-mcp-server))
+- Understanding of [MCP Authorization](https://modelcontextprotocol.io/specification/2025-06-18/basic/authorization)
+- An OAuth 2.1 compliant authorization server (e.g., WorkOS AuthKit, Auth0, Descope, etc.)
+- Authorization server's JWKS endpoint URL
+
+</GuideOverview.Prerequisites>
+
+<GuideOverview.YouWillLearn>
+
+- What Resource Server authentication is and why it's needed
+- How to configure your MCP server to validate OAuth tokens
+- How to support multiple authorization servers
+- How to use environment variables for production deployments
+
+</GuideOverview.YouWillLearn>
+</GuideOverview>
+
+## Understanding Resource Server Authentication
+
+### What is it?
+
+Resource Server authentication turns your MCP server into an [OAuth 2.1 Protected Resource](https://www.ietf.org/archive/id/draft-ietf-oauth-v2-1-13.html#name-roles) that validates Bearer tokens on every HTTP request. Your MCP server trusts one or more [Authorization Servers](https://www.ietf.org/archive/id/draft-ietf-oauth-v2-1-13.html#name-roles) to issue valid tokens for accessing your MCP server.
+
+### Why is it needed?
+
+By default, HTTP MCP servers cannot use tools that require authorization or secrets for security reasons.
+
+Resource Server authentication solves this by:
+
+1. **Authenticating every request** - Validates the Bearer token before processing any MCP messages
+2. **Extracting user identity** - The token's `sub` claim becomes the `context.user_id` for tool execution
+3. **Enabling secure tools** - Tools requiring authorization or secrets can now safely execute over HTTP
+4. **Supporting OAuth discovery** - MCP clients can automatically discover your authentication requirements
+
+<Callout type="info">
+**Resource Server auth vs Tool-level auth**: Resource Server authentication secures _access to your MCP server_, while tool-level authorization secures _access to third-party APIs that your tools use_. They work together: Resource Server auth identifies _who_ is calling your server, and tool-level auth enables tools to act _on behalf of that user_.
+</Callout>
+
+<Steps>
+
+## Choose Your Configuration Approach
+
+The `arcade_mcp_server.resource_server` module provides two validators:
+
+<Tabs items={["ResourceServer (Recommended)", "JWKSTokenValidator (Simple)"]}>
+<Tabs.Tab>
```

Comment:
I think we need to talk about how things work when you `arcade deploy` too - in that we handle all of this for you (always be selling).  Can `arcade deploy` be the first tab?

Reply:
updated